keras_cv/layers/preprocessing3d/random_drop_box.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing3d import base_augmentation_layer_3d
+from keras_cv.ops.point_cloud import is_within_box3d
+
+POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
+BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+BOX_LABEL_INDEX = base_augmentation_layer_3d.BOX_LABEL_INDEX
+
+
+class RandomDropBox(base_augmentation_layer_3d.BaseAugmentationLayer3D):
+    """A preprocessing layer which randomly drops object bounding boxes and points during training.
+
+    This layer will randomly drop object point clouds and bounding boxes. Number of dropped bounding boxes
+    is sampled uniformly sampled between 0 and max_drop_bounding_boxes. If label_index is set, only bounding boxes with
+    box class == label_index will be sampled and dropped; otherwise, all valid bounding boxes (box class > 0) will be sampled and dropped.
+
+    During inference time, the output will be identical to input. Call the layer with `training=True` to drop object bounding boxes and points.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 8 features are [x, y, z, dx, dy, dz, phi, box class].
+
+
+    Output shape:
+      A tuple of two Tensors (point_clouds, bounding_boxes) with the same shape as input Tensors.
+
+    Arguments:
+      label_index: An optional int scalar sets the target object index.
+        If label index is set, randomly drop bounding boxes, where box class == label_index.
+        If label index is None, randomly drop bounding boxes, where box class > 0.
+      max_drop_bounding_boxes: A int scalar sets the maximum number of dropped bounding boxes.
+
+    """
+
+    def __init__(self, label_index=None, max_drop_bounding_boxes=10, **kwargs):
+        super().__init__(**kwargs)
+        self.auto_vectorize = False
```

Comment:
nit: maybe we should just make auto_vectorize by default `False` for the moment.

Reply:
Done.